src/Adapter/Cart/CommandHandler/UpdateCartDeliverySettingsHandler.php

```diff
@@ -125,110 +150,99 @@ private function createCartRule(Cart $cart, string $backOfficeOrderCode): CartRu
     }
 
     /**
+     * This method works as follows:
+     * 1. if free shipping should be enabled, enable it
+     * 2. if free shipping should not be enabled and cart already does not have free shipping, do nothing
+     * 3.if free shipping should not be enabled and cart has free shipping, disable it
+     *
      * @param Cart $cart
      * @param UpdateCartDeliverySettingsCommand $command
      *
+     * @return bool should cart be saved after or not
+     *
      * @throws CannotDeleteCartRuleException
-     * @throws CartRuleException
-     * @throws PrestaShopException
      */
-    protected function handleFreeShippingOption(Cart $cart, UpdateCartDeliverySettingsCommand $command): void
+    protected function handleFreeShippingOption(Cart $cart, UpdateCartDeliverySettingsCommand $command): bool
     {
         $backOfficeOrderCode = sprintf('%s%s', CartRule::BO_ORDER_CODE_PREFIX, $cart->id);
 
         $freeShippingCartRule = $this->getCartRuleForBackOfficeFreeShipping($backOfficeOrderCode);
 
-        if ($command->allowFreeShipping()) {
+        $freeShippingShouldBeEnabled = $command->allowFreeShipping();
+
+        // Step 1
+        if ($freeShippingShouldBeEnabled) {
             if (null === $freeShippingCartRule) {
+                // there is not yet a 'free shipping' cart rule available in the system so we create it
                 $freeShippingCartRule = $this->createCartRule($cart, $backOfficeOrderCode);
             }
             $cart->addCartRule((int) $freeShippingCartRule->id);
 
-            return;
+            return true;
         }
 
         if (null === $freeShippingCartRule) {
-            return;
+            // Step 2
+            return false;
         }
 
+        // Step 3
```

Comment:
Be careful, you confused this one with the three others The three others required update the cart
But this one performed a delete on the cart rule, so it should be handled independently without managing the return as it doesn't require the cart update